src/Uno.UI/Microsoft/UI/Xaml/Controls/ColorPicker/ColorSpectrumAutomationPeer.cs

```diff
@@ -0,0 +1,142 @@
+{
+	public class ColorSpectrumAutomationPeer : AutomationPeer, IValueProvider
+	{
+		// Uno Doc: Added for the Uno Platform
+		private readonly ColorSpectrum _owner;
+
+		internal ColorSpectrumAutomationPeer(ColorSpectrum owner)
+		{
+			_owner = owner;
+		}
+
+		// IAutomationPeerOverrides 
+		protected override object GetPatternCore(PatternInterface patternInterface)
+		{
+			if (patternInterface == PatternInterface.Value)
+			{
+				return this;
+			}
+
+			return base.GetPatternCore(patternInterface);
+		}
+
+		protected override AutomationControlType GetAutomationControlTypeCore()
+		{
+			return AutomationControlType.Slider;
+		}
+
+		protected override string GetLocalizedControlTypeCore()
+		{
+			return ResourceAccessor.GetLocalizedStringResource(ResourceAccessor.SR_LocalizedControlTypeColorSpectrum);
+		}
+
+		protected override string GetNameCore()
+		{
+			string nameString = base.GetNameCore();
+
+			// If a name hasn't been provided by AutomationProperties.Name in markup,
+			// then we'll return the default value.
+			if (string.IsNullOrEmpty(nameString))
+			{
+				nameString = ResourceAccessor.GetLocalizedStringResource(ResourceAccessor.SR_AutomationNameColorSpectrum);
+			}
+
+			return nameString;
+		}
+
+		protected override string GetClassNameCore()
+		{
+			return nameof(ColorSpectrum);
+		}
+
+		protected override string GetHelpTextCore()
+		{
+			return ResourceAccessor.GetLocalizedStringResource(ResourceAccessor.SR_HelpTextColorSpectrum);
+		}
+
+		protected override Rect GetBoundingRectangleCore()
+		{
+			Rect boundingRectangle = _owner.GetBoundingRectangle();
+			return boundingRectangle;
+		}
+
+		protected override Point GetClickablePointCore()
+		{
+			var boundingRect = GetBoundingRectangleCore(); // Call potentially overridden method
+
+			return new Point(boundingRect.X + boundingRect.Width / 2, boundingRect.Y + boundingRect.Height / 2);
+		}
+
+		// IValueProvider properties and methods
+		public bool IsReadOnly
+		{
+			get => false;
+		}
+
+		public string Value
+		{
+			get 
+			{
+				ColorSpectrum colorSpectrumOwner = _owner;
+				Color color = colorSpectrumOwner.Color;
+				Vector4 hsvColor = colorSpectrumOwner.HsvColor;
+
+				return GetValueString(color, hsvColor);
+			}
+		}
+
+		public void SetValue(string value)
+		{
+			ColorSpectrum colorSpectrumOwner = _owner;
+			Color color = (Color)(XamlBindingHelper.ConvertValue(typeof(Color), value));
+
+			colorSpectrumOwner.Color = color;
+
+			// Since ColorPicker sets ColorSpectrum.Color and ColorPicker also responds to ColorSpectrum.ColorChanged,
+			// we could get into an infinite loop if we always raised ColorSpectrum.ColorChanged when ColorSpectrum.Color changed.
+			// Because of that, we'll raise the event manually.
+			colorSpectrumOwner.RaiseColorChanged();
+		}
+
+		public void RaisePropertyChangedEvent(Color oldColor, Color newColor, Vector4 oldHsvColor, Vector4 newHsvColor)
+		{
+			string oldValueString = GetValueString(oldColor, oldHsvColor);
```

Comment:
Codacy found an issue: [Remove the unused local variable 'oldValueString'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6121259)